logoapp/js/dot_data.js

```diff
@@ -0,0 +1,1028 @@
+			var endAngle = 360/num * (i+0.5) * Math.PI / 180;
+			context.arc(this.pos.x + translate.x, this.pos.y+translate.y, this.r*1.5, startAngle, endAngle, false);
+			context.stroke();
+		}
+	}
+};
+Dot.prototype.isPointOnDot = function(point, range, center) {
+	var xl = this.pos.x+center.y-point.x;
+	var yl = this.pos.y+center.x-point.y;
+	var l2 = (xl*xl) + (yl*yl);
+	var r2 = (this.r+range)*(this.r+range);
+	if(l2 < r2) return true;
+	else return false;
+};
+Dot.prototype.copy = function() {
+	var newDot = new Dot(new Vector2(this.pos.x, this.pos.y), this.r, this.color, this.isDraw);
+	return newDot;
+};
+
+DotFrame = function(interval, radius, bgColor) {
+	this.bgColor = bgColor;
+	var dotData = new Array(DOT_NUM);
+	for(var i=0; i<dotData.length; i++) {
+		dotData[i] = new Dot(getDotPos(i, interval), radius, '#000', false);
+	}
+	this.arrDot = dotData;
+};
+DotFrame.prototype.drawMain = function(view, isHoverDot, hoverDotNum, isPlay) {
+	var center = new Vector2(ORIGINAL_CANVAS_SIZE/2, ORIGINAL_CANVAS_SIZE/2);
+	if(!isPlay && isHoverDot) {
+		this.arrDot[hoverDotNum].drawHover(view.ctx, center);
+	}
+	for(var i=0; i<this.arrDot.length; i++) {
+		if(!isHoverDot || i!=hoverDotNum) {
+			this.arrDot[i].drawMain(view.ctx, center, isPlay);
+		}
+	}
+};
+DotFrame.prototype.drawPreview = function(view1) {
+	var center = new Vector2(ORIGINAL_CANVAS_SIZE/2, ORIGINAL_CANVAS_SIZE/2);
+	for(var i=0; i<this.arrDot.length; i++) {
+		this.arrDot[i].drawPreview(view1.ctx, center);
+	}
+};
+DotFrame.prototype.copy = function() {
+	var newDotFrame = new DotFrame(0, 0, '#000');
+	newDotFrame.bgColor = this.bgColor;
+	for(var i=0; i<DOT_NUM; i++) {
+		newDotFrame.arrDot[i] = this.arrDot[i].copy();
+	}
+	return newDotFrame;
+};
+
+DotAppFramework = function(mainCanvasName, previewCanvasName) {
+	this.mainView = new View(mainCanvasName, MAINCANVAS_WIDTH, MAINCANVAS_HEIGHT, true);
+	this.mainView.ctx.scale(MAIN_SCALE, MAIN_SCALE);
+	var arrPreview = new Array(8);
+	for(var i=0; i<arrPreview.length; i++) {
+		arrPreview[i] = new View(previewCanvasName+""+(i+1), PREVIEWCANVAS_WIDTH, PREVIEWCANVAS_HEIGHT, false);
+	}
+	this.previewView = arrPreview;
+	this.mainView.canvas.addEventListener('click', this.onClickMain.bind(this), false);
+	this.mainView.canvas.addEventListener('mousedown', this.onMouseDownMain.bind(this), false);
+	this.mainView.canvas.addEventListener('mouseup', this.onMouseUpMain.bind(this), false);
+	for(var i=0; i<this.previewView.length; i++) {
+		this.previewView[i].canvas.addEventListener('click', this.onClickPreview.bind(this), false);
+		this.previewView[i].ctx.scale(PREVIEW_SCALE, PREVIEW_SCALE);
+	}
+	this.completeView = new View('completeCanvas', MAINCANVAS_WIDTH, MAINCANVAS_HEIGHT, false);
+	this.completeView.ctx.scale(MAIN_SCALE, MAIN_SCALE);
+	this.isCompleteView = false;
+	this.completeViewFrame = 1;
+	this.completeViewTime = 0;
+	this.completeLoopInterval;
+	var frameData = new Array(8);
+	for(var i=0; i<frameData.length; i++) {

```

Comment:
__[jshint]__
*  'i' is already defined.


<!-- Comment by [SideCI](https://sideci.com) -->